src/client/pong3.jsx

```diff
@@ -1,3 +1,52 @@
-const RoomSelector = require('./RoomSelector.js');
+import ReactDOM from 'react-dom';
+import React, {useState, useEffect, useRef} from 'react';
+import RoomSelector from './RoomSelector';
+import ioClient from 'socket.io-client/dist/socket.io';
+import Game from './Game';
 
-const selector = new RoomSelector(document.querySelector('.room-selector'));
+const App = () => {
+  const [game, setGame] = useState(null);
+  const gameObj = useRef({game, setGame});
+  const [socket, setSocket] = useState(null);
+  const [joinRoom, setJoinRoom] = useState(() => null);
+
+  gameObj.current.game = game;
+  gameObj.current.setGame = setGame;
+  useEffect(() => {
+    const socket = ioClient.connect();
+    const joinRoom = async (roomId) => {
+      socket.emit('joinRoom', roomId);
+    };
+
+    socket.on('joinedRoom', () => {
+      console.log('joined room');
+      const startGame = data => {
+        console.log('game started');
+        gameObj.current.setGame(new Game(data.controlledPlayer, data.initialBall, socket));
```

Comment:
have to use a ref here because otherwise, game and setGame always have their values from the first render, which prevents us from setting the game or destroying it.